app/src/main/java/org/vss/impl/postgres/PostgresBackendImpl.java

```diff
@@ -0,0 +1,131 @@
+package org.vss.impl.postgres;
+
+import com.google.inject.Inject;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Singleton;
+import org.jooq.DSLContext;
+import org.jooq.Insert;
+import org.jooq.Query;
+import org.jooq.Update;
+import org.vss.GetObjectRequest;
+import org.vss.GetObjectResponse;
+import org.vss.KVStore;
+import org.vss.KeyValue;
+import org.vss.ListKeyVersionsRequest;
+import org.vss.ListKeyVersionsResponse;
+import org.vss.PutObjectRequest;
+import org.vss.PutObjectResponse;
+import org.vss.exception.ConflictException;
+import org.vss.postgres.tables.records.VssDbRecord;
+
+import static org.jooq.impl.DSL.val;
+import static org.vss.postgres.tables.VssDb.VSS_DB;
+
+@Singleton
+public class PostgresBackendImpl implements KVStore {
+
+  private final DSLContext context;
+
+  @Inject
+  public PostgresBackendImpl(DSLContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public GetObjectResponse get(GetObjectRequest request) {
+
+    VssDbRecord vssDbRecord = context.selectFrom(VSS_DB)
+        .where(VSS_DB.STORE_ID.eq(request.getStoreId())
+            .and(VSS_DB.KEY.eq(request.getKey())))
+        .fetchOne();
+
+    if (vssDbRecord != null) {
+      return GetObjectResponse.newBuilder()
+          .setValue(KeyValue.newBuilder()
+              .setKey(vssDbRecord.getKey())
+              .setValue(ByteString.copyFrom(vssDbRecord.getValue()))
+              .setVersion(vssDbRecord.getVersion())
+              .build())
+          .build();
+    } else {
+      return GetObjectResponse.newBuilder()
+          .setValue(KeyValue.newBuilder()
+              .setKey(request.getKey()).build())
+          .build();
+    }
+  }
+
+  @Override
+  public PutObjectResponse put(PutObjectRequest request) {
+
+    String storeId = request.getStoreId();
+
+    List<VssDbRecord> vssRecords = new ArrayList<>(request.getTransactionItemsList().stream()
+        .map(kv -> buildVssRecord(storeId, kv)).toList());
+
+    VssDbRecord globalVersionRecord = buildVssRecord(storeId,
+        KeyValue.newBuilder()
+            .setKey(GLOBAL_VERSION_KEY)
```

Comment:
I think the global version should be stored in a separate table.  otherwise, you have to make sure the caller doesn't try to use this key, and it seems messy.

Reply:
The advantage is that it can be treated as any other keyvalue and easily handled with exactly same cases. 

Other consideration is, if we want client to be able to override global_version in certain scenarios. My take is we should allow, end of the day its their storage and versioning, they can do whatever they want with it.

There is also case where client just wants to do a "get" on global_version. 
For all purposes, it seems simpler to keep it as normal key-value, otherwise i will need to introduce additional code to support everything.

One case that we will have to handle separately from normal key-value is to not return global_version as part of ListKeyVersions api. (we have dedicated field in response for it)